x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/transport/SSLExceptionHelper.java

```diff
@@ -22,6 +24,11 @@ public static boolean isNotSslRecordException(Throwable e) {
     }
 
     public static boolean isCloseDuringHandshakeException(Throwable e) {
+        return isCloseDuringHandshakeSSLException(e)
+                || isCloseDuringHandshakeSSLException(e.getCause());
```

Comment:
We were already checking for `Received close_notify during handshake`, just not within a `DecoderException`. Not sure whether this was a mistake or whether we do sometimes see this exception unwrapped too.

Also not sure what the implications for the `nio` transport since the `DecoderException` wrapper is Netty-specific.